src/peer_digest.cc

```diff
@@ -705,26 +662,16 @@ peerDigestReqFinish(DigestFetchState * fetch, char * /* buf */,
             pd->times.retry_delay = 0;
             peerDigestSetCheck(pd, peerDigestNewDelay(fetch->entry));
         }
+        pd->finish(fetch, err);
     }
 
-    /* note: order is significant */
-    if (fcb_valid)
-        peerDigestFetchSetStats(fetch);
-
-    if (pdcb_valid)
-        peerDigestPDFinish(fetch, pcb_valid, err);
-
-    if (fcb_valid)
-        peerDigestFetchFinish(fetch, err);
+    peerDigestFetchFinish(fetch, err);
 }
 
-/* destroys digest if peer disappeared
- * must be called only when fetch and pd cbdata are valid */
-static void
-peerDigestPDFinish(DigestFetchState * fetch, int pcb_valid, int err)
+void
+PeerDigest::finish(DigestFetchState * const fetch, const int err)
 {
-    PeerDigest *pd = fetch->pd;
-    const auto host = pd->host;
+    const auto pd = this; // TODO: remove this diff reducer
```

Comment:
```suggestion
    const auto pd = this; // TODO: remove this diff reducer
    const auto fetch = &finishedFetch; // TODO: remove this diff reducer
```

... and adjust the first parameter to be a DigestFetchState reference. A constant reference is possible, of course.

Reply:
Done.